### ISSUE TITLE
Possible documentation error for setting up collection preview thumbnail.

### ISSUE THREAD

**Issue by michaeldzjap** [spatie/laravel-medialibrary#3464]
I've installed v10.15 and following the [documentation](https://spatie.be/docs/laravel-medialibrary/v10/basic-usage/preparing-your-model) I'm adding the method

```php
public function registerMediaConversions(?Media $media = null): void
{
    $this
         ->addMediaConversion('preview')
         ->fit(Manipulations::FIT_CROP, 300, 300)
         ->nonQueued();
}
```

to my model. However, running _PHPStan_ results in the error:

> Call to an undefined method Spatie\Image\Manipulations::nonQueued().

It seems the order of the function calls (no longer) works. Changing it to this:

```php
$this
    ->addMediaConversion('preview')
    ->nonQueued()
    ->getManipulations()
    ->fit(Manipulations::FIT_CROP, 300, 300);
```

gets rid of the error. The documentation possibly needs to be updated?

**Comment by freekmurze** [comment]
This is not really an error, PHPStan gets confused by the magic that is happening to make this fluent API work. Best to add this error to your baseline.

**Comment by michaeldzjap** [comment]
Personally, I'm not a fan of this sort of "magic". It makes logic unnecessarily hard to follow in my opinion, but OK, I can't really complain about that if I decide someone else's package. I did notice though that the example as its currently presented in the documentation still doesn't work. It seems _spatie/image_ has removed the `\Spatie\Image\Manipulations` class. Would this:

```php
$this
    ->addMediaConversion('preview')
    ->nonQueued()
    ->getManipulations()
    ->addManipulation('fit', [\Spatie\Image\Enums\Fit::Crop, 300, 300]);
```

do the same without any magic calls and the correct replacement for `\Spatie\Image\Manipulations::FIT_CROP`?
